aws-geo-location/src/main/java/com/amplifyframework/geo/location/configuration/GeoConfiguration.kt

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.geo.location.configuration
+
+import com.amplifyframework.geo.location.AWSLocationGeoPlugin
+import org.json.JSONObject
+
+/**
+ * Configuration options for [AWSLocationGeoPlugin].
+ */
+data class GeoConfiguration internal constructor(
+    val region: String,
+    val maps: MapsConfiguration?
+) {
+    companion object {
+        private const val DEFAULT_REGION = "us-east-1"
+
+        /**
+         * Returns a builder object for plugin configuration.
+         * @return fresh configuration builder instance.
+         */
+        fun builder(): Builder {
```

Comment:
I think you may need to make use of the `@JvmStatic` annotation here. https://developer.android.com/kotlin/interop#companion_functions

I was looking through the interop guide and if I'm reading it correct, java callers would have to use `GeoConfiguration.Compation.builder()` if you don't use `@JvmStatic`.